object/slicer/slicer.go

```diff
@@ -76,82 +82,107 @@ type Slicer struct {
 // are written to the parameterized ObjectWriter. If the amount of data is
 // within the limit, one object is produced. Note that Slicer can write multiple
 // objects, but returns the root object ID only.
-//
-// If current NeoFS epoch is specified via Options.SetCurrentNeoFSEpoch, it is
-// written to the metadata of all resulting objects as a creation epoch.
-//
-// See also NewSession.
-func New(signer neofscrypto.Signer, cnr cid.ID, owner user.ID, w ObjectWriter, opts Options) *Slicer {
+func New(ctx context.Context, nw NetworkedClient, signer neofscrypto.Signer, cnr cid.ID, owner user.ID, sessionToken *session.Object) (*Slicer, error) {
+	ni, err := nw.NetworkInfo(ctx, client.PrmNetworkInfo{})
+	if err != nil {
+		return nil, fmt.Errorf("network info: %w", err)
+	}
+
+	opts := Options{
+		objectPayloadLimit: ni.MaxObjectSize(),
+		currentNeoFSEpoch:  ni.CurrentEpoch(),
+		sessionToken:       sessionToken,
+	}
+
+	if !ni.HomomorphicHashingDisabled() {
+		opts.CalculateHomomorphicChecksum()
+	}
+
+	var hdr object.Object
+	hdr.SetContainerID(cnr)
+	hdr.SetType(object.TypeRegular)
+	hdr.SetOwnerID(&owner)
+	hdr.SetCreationEpoch(ni.CurrentEpoch())
+	hdr.SetSessionToken(sessionToken)
+
 	return &Slicer{
-		signer: signer,
-		cnr:    cnr,
-		owner:  owner,
-		w:      w,
 		opts:   opts,
-	}
+		w:      nw,
+		signer: signer,
+		hdr:    hdr,
+	}, nil
 }
 
-// NewSession creates Slicer which generates objects within provided session.
-// NewSession work similar to New with the detail that the session issuer owns
-// the produced objects. Specified session token is written to the metadata of
-// all resulting objects. In this case, the object is considered to be created
-// by a proxy on behalf of the session issuer.
-func NewSession(signer neofscrypto.Signer, cnr cid.ID, token session.Object, w ObjectWriter, opts Options) *Slicer {
-	return &Slicer{
-		signer:       signer,
-		cnr:          cnr,
-		owner:        token.Issuer(),
-		w:            w,
-		opts:         opts,
-		sessionToken: &token,
-	}
+// Put creates new NeoFS object from the input data stream, associates the
+// object with the configured container and writes the object via underlying
+// [ObjectWriter]. After a successful write, Slice returns an [oid.ID] which is a
+// unique reference to the object in the container. Slice sets all required
+// calculated fields like payload length, checksum, etc.
+//
+// Slice allows you to specify [object.Attribute] parameters to be written to the
+// resulting object's metadata. Keys SHOULD NOT start with system-reserved
+// '__NEOFS__' prefix.
+//
+// See [New] for details.
+func (x *Slicer) Put(data io.Reader, attrs []object.Attribute) (oid.ID, error) {
```

Comment:
`context.Context`?

Reply:
Added context to all methods

Reply:
Why do you need it in a struct? It's always a single-call thing.